### PR TITLE
JDK-8296287: Improve documentation for Types.directSupertypes()

### PR DESCRIPTION
**Pull request by jddarcy**
Please review the PR and CSR (https://bugs.openjdk.org/browse/JDK-8296354) to add text to explicitly specify the behavior of Types.directSupertypes on java.lang.Object.

(This text could be added as informative text as it is implied by the existing specification, but I think it was reasonable to add the text as a normative statement instead.)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8296287](https://bugs.openjdk.org/browse/JDK-8296287): Improve documentation for Types.directSupertypes()
 * [JDK-8296354](https://bugs.openjdk.org/browse/JDK-8296354): Improve documentation for Types.directSupertypes() (**CSR**)

### Reviewers
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**) ⚠️ Review applies to [e216112e](https://git.openjdk.org/jdk/pull/10977/files/e216112eeca338fb635987d175c85d5a7f40643d)
 * [Pavel Rappo](https://openjdk.org/census#prappo) (@pavelrappo - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10977/head:pull/10977` \
`$ git checkout pull/10977`

Update a local copy of the PR: \
`$ git checkout pull/10977` \
`$ git pull https://git.openjdk.org/jdk pull/10977/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10977`

View PR using the GUI difftool: \
`$ git pr show -t 10977`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10977.diff">https://git.openjdk.org/jdk/pull/10977.diff</a>

</details>

### CODE REVIEW

File: test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Verify java.lang.Object has an empty list of direct supertypes.
+ */
+public class TestDirectSupertypeObject extends JavacTestingAbstractProcessor {
+    public boolean process(Set<? extends TypeElement> annotations,
+                           RoundEnvironment roundEnv) {
+        if (!roundEnv.processingOver()) {
+            TypeMirror objectType  = requireNonNull(eltUtils.getTypeElement("java.lang.Object")).asType();
+            var ojectSupertypes = typeUtils.directSupertypes(objectType);
```
Comment (lahodaj) [inline test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java:48]: Nit: `objectSupertypes` -> `objectSupertypes`.

**Review by lahodaj — APPROVED** [review]
Looks good to me.

File: test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java (hunk 1 above)
Comment (pavelrappo) [reply test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java:48]: _ojectSupertypes_

File: test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8296287
+ * @summary Test Types methods on module and package TypeMirrors
```
Comment (pavelrappo) [inline test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java:27]: I don't think I understand the summary. Specifically, how are modules and packages pertinent to this test? (I'm not an expert in compiler area.)

File: test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import static java.util.Objects.*;
+
+/**
+ * Verify java.lang.Object has an empty list of direct supertypes.
+ */
+public class TestDirectSupertypeObject extends JavacTestingAbstractProcessor {
+    public boolean process(Set<? extends TypeElement> annotations,
+                           RoundEnvironment roundEnv) {
+        if (!roundEnv.processingOver()) {
+            TypeMirror objectType  = requireNonNull(eltUtils.getTypeElement("java.lang.Object")).asType();
```
Comment (pavelrappo) [inline test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java:47]: Redundant whitespace before =

File: test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java (hunk 1 above)
Comment (jddarcy) [reply test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java:48]: Thanks; will fix before pushing.

File: test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java (hunk 2 above)
Comment (jddarcy) [reply test/langtools/tools/javac/processing/model/util/types/TestDirectSupertypeObject.java:27]: Ah, that was a left over from copying the other test in that directory; will correct to an accurate summary.

**Review by pavelrappo — APPROVED** [review]
Looks good.

**Pull request closed without merging**